agent/exec/plugin/adapter.go

```diff
@@ -0,0 +1,48 @@
+package plugin
+
+import (
+	"github.com/docker/docker/api/types"
+	engineapi "github.com/docker/docker/client"
+	"github.com/docker/swarmkit/api"
+	"golang.org/x/net/context"
+)
+
+type pluginAdapter struct {
+	client engineapi.APIClient
+	plugin *pluginConfig
+}
+
+func newPluginAdapter(client engineapi.APIClient, task *api.Task) (*pluginAdapter, error) {
+	plgn, err := newPluginConfig(task)
```

Comment:
I kept `plgn ` to be consistent with the existing `container` controller using `ctr`